### PR TITLE
Refactor routing package 

### PR DESCRIPTION
**Pull request by arberiii**
As described in [#872](https://github.com/short-d/short/issues/872), the routing package needs refactoring.

In this PR I moved the handlers to a package of their own with the idea that they should be alone since routing should be responsible only for routing.

### CODE REVIEW

File: backend/app/adapter/routing/route.go [hunk 1]
```diff
@@ -34,83 +39,67 @@ func NewShort(
...
-			Handle: SearchHandle(search),
-		},
-	}
+
+	var shortRouter shortRouter
+
+	shortRouter.addRoute("GET", "/oauth/github/sign-in", handle.NewSSOSignIn(
+		sso.SingleSignOn(githubSSO),
+		webFrontendURL,
+	))
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:48]: It does look more concise. However, we can't see the parameter name any more.

File: backend/app/adapter/routing/route.go [hunk 2]
```diff
@@ -34,83 +39,67 @@ func NewShort(
...
+		shortLinkRetriever,
+		timer,
+		*frontendURL,
+	))
+	shortRouter.addRoute("GET", "/features/:featureID", handle.Feature(
+		instrumentationFactory,
+		featureDecisionMakerFactory,
+		authenticator,
+	))
+	shortRouter.addRoute("GET", "/analytics/track/:event", analytics.TrackHandle(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:88]: Let's move `TrackHandle` into handle!

Comment (arberiii) [reply backend/app/adapter/routing/route.go:88]: I agree!

File: backend/app/adapter/routing/route.go (hunk 1 above)
Comment (arberiii) [reply backend/app/adapter/routing/route.go:48]: I will revert this part

File: backend/app/adapter/routing/route.go [hunk 3]
```diff
@@ -38,55 +38,55 @@ func NewShort(
...
+			Handle: handle.NewSSOSignInCallback(
 				sso.SingleSignOn(googleSSO),
 				*frontendURL,
 			),
 		},
 		{
 			Method: "GET",
 			Path:   "/r/:alias",
-			Handle: NewLongLink(
+			Handle: handle.NewLongLink(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:89]:
```suggestion
			Handle: handle.LongLink(
```

File: backend/app/adapter/routing/route.go [hunk 4]
```diff
@@ -38,55 +38,55 @@ func NewShort(
...
+			Handle: handle.NewSSOSignIn(
 				sso.SingleSignOn(googleSSO),
 				webFrontendURL,
 			),
 		},
 		{
 			Method: "GET",
 			Path:   "/oauth/google/sign-in/callback",
-			Handle: NewSSOSignInCallback(
+			Handle: handle.NewSSOSignInCallback(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:81]:
```suggestion
			Handle: handle.SSOSignInCallback(
```

File: backend/app/adapter/routing/route.go [hunk 5]
```diff
@@ -38,55 +38,55 @@ func NewShort(
...
 				sso.SingleSignOn(facebookSSO),
 				*frontendURL,
 			),
 		},
 		{
 			Method: "GET",
 			Path:   "/oauth/google/sign-in",
-			Handle: NewSSOSignIn(
+			Handle: handle.NewSSOSignIn(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:73]:
```suggestion
			Handle: handle.SSOSignIn(
```

File: backend/app/adapter/routing/route.go [hunk 6]
```diff
@@ -38,55 +38,55 @@ func NewShort(
...
 				sso.SingleSignOn(facebookSSO),
 				webFrontendURL,
 			),
 		},
 		{
 			Method: "GET",
 			Path:   "/oauth/facebook/sign-in/callback",
-			Handle: NewSSOSignInCallback(
+			Handle: handle.NewSSOSignInCallback(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:65]:
```suggestion
			Handle: handle.SSOSignInCallback(
```

File: backend/app/adapter/routing/route.go [hunk 7]
```diff
@@ -38,55 +38,55 @@ func NewShort(
...
 				sso.SingleSignOn(githubSSO),
 				*frontendURL,
 			),
 		},
 		{
 			Method: "GET",
 			Path:   "/oauth/facebook/sign-in",
-			Handle: NewSSOSignIn(
+			Handle: handle.NewSSOSignIn(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:57]:
```suggestion
			Handle: handle.SSOSignIn(
```

File: backend/app/adapter/routing/route.go [hunk 8]
```diff
@@ -38,55 +38,55 @@ func NewShort(
...
 				sso.SingleSignOn(githubSSO),
 				webFrontendURL,
 			),
 		},
 		{
 			Method: "GET",
 			Path:   "/oauth/github/sign-in/callback",
-			Handle: NewSSOSignInCallback(
+			Handle: handle.NewSSOSignInCallback(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:49]:
```suggestion
			Handle: handle.SSOSignInCallback(
```

File: backend/app/adapter/routing/route.go [hunk 9]
```diff
@@ -38,55 +38,55 @@ func NewShort(
 		{
 			Method: "GET",
 			Path:   "/oauth/github/sign-in",
-			Handle: NewSSOSignIn(
+			Handle: handle.NewSSOSignIn(
```
Comment (byliuyang) [inline backend/app/adapter/routing/route.go:41]:
```suggestion
			Handle: handle.SSOSignIn(
```

**Pull request merged**